File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/cost/model/CostInfo.java
package com.study.webflux.rag.domain.cost.model;

/** LLM/TTS와 전체 크레딧 비용을 포함한 대화 비용 정보입니다. */
public record CostInfo(
	long llmCredits,
	long ttsCredits,
	long totalCredits
) {
	public static CostInfo of(long llmCredits, long ttsCredits) {
		return new CostInfo(llmCredits, ttsCredits, llmCredits + ttsCredits);
	}

	public static CostInfo zero() {
		return new CostInfo(0, 0, 0);
	}
}

File: backend/domain/src/main/kotlin/com/miyou/app/domain/cost/model/CostInfo.kt
package com.miyou.app.domain.cost.model

/**
 * LLM/TTS 비용 계산 결과를 보관하는 값 객체.
 */
data class CostInfo(
    val llmCredits: Long,
    val ttsCredits: Long,
    val totalCredits: Long,
) {
    companion object {
        @JvmStatic
        fun of(
            llmCredits: Long,
            ttsCredits: Long,
        ): CostInfo = CostInfo(llmCredits, ttsCredits, llmCredits + ttsCredits)

        @JvmStatic
        fun zero(): CostInfo = CostInfo(0L, 0L, 0L)
    }
}
